### PR TITLE
Support 'meta' as key in meta dictionary

### LINKED ISSUE #256: Cannot load database if meta contains key 'meta'
Minimal example that fails:

```python
db = audformat.testing.create_db(minimal=True)
db.meta['meta'] = 'value'
db.save('db')
audformat.Database.load('db')
```
```
Traceback (most recent call last):
  File "/media/jwagner/Data/Git/pyaudformat/my/my.py", line 24, in <module>
    audformat.Database.load('db')
  File "/media/jwagner/Data/Git/pyaudformat/audformat/core/database.py", line 917, in load
    db = Database.load_header_from_yaml(header)
  File "/media/jwagner/Data/Git/pyaudformat/audformat/core/database.py", line 977, in load_header_from_yaml
    db.from_dict(header, ignore_keys=['media', 'misc_tables', 'raters',
  File "/media/jwagner/Data/Git/pyaudformat/audformat/core/common.py", line 191, in from_dict
    assert isinstance(value, type(self.__dict__[key]))
AssertionError
```

### PR DESCRIPTION
**Pull request by frankenjoe**
Closes #256

### CODE REVIEW

File: tests/test_common.py [hunk 1]
```diff
@@ -3,6 +3,72 @@
...
+    for item, expected in zip(d, ['a', 'b', 'c']):
+        assert item == expected
+    for item, expected in zip(reversed(d), ['c', 'b', 'a']):
+        assert item == expected
+    assert d.popitem(last=True) == ('c', 2)
+    assert d.popitem(last=False) == ('a', 0)
+    assert list(d) == ['b']
+
+
+def test_meta_dict():
```
Comment (frankenjoe) [inline tests/test_common.py:62]: Here's the new test I added, the other tests I just moved around to order them alphabetically.

File: tests/test_common.py [hunk 2]
```diff
@@ -3,6 +3,72 @@
...
+    meta = {
+        'key': 'value',
+        'meta': 1234,
+    }
+    header = audformat.core.common.HeaderBase(meta=meta)
```
Comment (hagenw) [inline tests/test_common.py:67]: To be more obvious that it also contains the correct value, maybe add:

```suggestion
    header = audformat.core.common.HeaderBase(meta=meta)
    assert header.meta['meta'] == 1234
```

File: tests/test_common.py [hunk 3]
```diff
@@ -3,6 +3,72 @@
...
+    d = header.to_dict()
+    header.from_dict(d)
```
Comment (hagenw) [inline tests/test_common.py:69]: ... and here as well

```suggestion
    header.from_dict(d)
    assert header.meta['meta'] == 1234
```

File: tests/test_common.py (hunk 2 above)
Comment (frankenjoe) [reply tests/test_common.py:74]: I locally updated the test

File: tests/test_common.py (hunk 3 above)
Comment (frankenjoe) [reply tests/test_common.py]: I locally updated the test

**Pull request merged**